sound/soc/sof/sof-audio.c

```diff
@@ -838,9 +838,11 @@ int sof_pcm_stream_free(struct snd_sof_dev *sdev, struct snd_pcm_substream *subs
 
 	if (spcm->prepared[substream->stream]) {
 		/* stop DMA first if needed */
-		if (pcm_ops && pcm_ops->platform_stop_during_hw_free)
+		if (pcm_ops && pcm_ops->platform_stop_during_hw_free &&
+		    spcm->delayed_stop[substream->stream]) {
 			snd_sof_pcm_platform_trigger(sdev, substream, SNDRV_PCM_TRIGGER_STOP);
-
+			spcm->delayed_stop[substream->stream] = false;
+		}
```

Comment:
well that's just fantastic @ranj063, this patch alone breaks suspend-resume-with-audio....